float32[] laser_scan
float32[] goal_in_robot_frame
float32[] last_action